src/Avalonia.Visuals/Media/FontFamily.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Avalonia.Media.Fonts;
+
+namespace Avalonia.Media
+{
+    public class FontFamily : IFontFamily
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:Avalonia.Media.FontFamily" /> class.
+        /// </summary>
+        /// <param name="name">The name of the <see cref="FontFamily"/>.</param>
+        /// <exception cref="T:System.ArgumentNullException">name</exception>
+        public FontFamily(string name)
+        {
+            Contract.Requires<ArgumentNullException>(name != null);
+
+            FamilyNames = new FamilyNameCollection(new[] { name });
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:Avalonia.Media.FontFamily" /> class.
+        /// </summary>
+        /// <param name="names">The names of the <see cref="FontFamily"/>.</param>
+        /// <exception cref="T:System.ArgumentNullException">name</exception>
+        public FontFamily(IEnumerable<string> names)
+        {
+            Contract.Requires<ArgumentNullException>(names != null);
+
+            FamilyNames = new FamilyNameCollection(names);
+        }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:Avalonia.Media.FontFamily" /> class.
+        /// </summary>
+        /// <param name="name">The name of the <see cref="T:Avalonia.Media.FontFamily" />.</param>
+        /// <param name="source">The source of font resources.</param>
+        public FontFamily(string name, Uri source) : this(name)
+        {
+            Key = new FontFamilyKey(source);
+        }
+
+        public static FontFamily Default => new FontFamily("Courier New");
+
+        public string Name => FamilyNames.PrimaryFamilyName;
+      
+        FamilyNameCollection IFontFamily.FamilyNames => FamilyNames;
```

Comment:
I don't understand this: `IFontFamily` exposes `Name`, `FamilyNames` and `Key` which are implemented here as explicit interface methods and forwarded to `internal` properties? Why not just make these properties public? That would have the same effect, no?

And why is `IFontFamily` needed in that case? It doesn't look like there can be any other implementation of it, as the `TextBlock.FontFamily` property is of type `FontFamily`. What use-case would having another implementation of `FontFamily` solve?

Reply:
Have removed the interface. It was ment to hide FamilyNames and Key for 3rd Party. The internal properties just made it easier to access them without casting.